src/AWS.Deploy.Orchestration/DeploymentSettingsHandler.cs

```diff
@@ -91,5 +100,53 @@ public async Task ApplySettings(DeploymentSettings deploymentSettings, Recommend
 
             throw new InvalidDeploymentSettingsException(DeployToolErrorCode.DeploymentConfigurationNeedsAdjusting, errorMessage.Trim());
         }
+
+        public async Task<string> SaveSettings(string filePath, Recommendation recommendation, CloudApplication cloudApplication, OrchestratorSession orchestratorSession)
+        {
+            var parentDirectory = Path.GetDirectoryName(filePath);
+            if (string.IsNullOrEmpty(parentDirectory))
+            {
+                return $"Failed to save deployment settings because {filePath} is not a valid path";
+            }
+            if (!_directoryManager.Exists(parentDirectory))
+            {
+                return $"Failed to save deployment settings because part of {filePath} does not exist on disk";
+            }
+
+            var deploymentSettings = new DeploymentSettings
```

Comment:
Doesn't the caller of `SaveSettings` (`DeployCommand.ExecuteAsync`) already have a `DeploymentSettings` object? Should we reuse that instead of recreating here?

I'm not sure if there's a good reason not to, just thinking about if we ever change how it's initialized there are now multiple places to update.

Reply:
The `DeploymentSettings` from `DeployCommand.ExecuteAsync` is only intialized if the `--apply` flag is provided.
https://github.com/aws/aws-dotnet-deploy/blob/dev/src/AWS.Deploy.CLI/Commands/CommandFactory.cs#L196-#L200

That object is only used to apply the deployment settings that were specified in a config file.